doc/10-Channels.md

```diff
@@ -1,88 +1,298 @@
 # Channels
 
 After Icinga Notifications decides to send a notification of any kind, it will be passed to a channel plugin.
-Such a channel plugin submits the notification event to a channel, e.g., email or a chat client.
+Such a plugin submits the notification event to a domain-specific channel, e.g., email or a chat client.
 
-Icinga Notifications comes packed with channel plugins, but also enables you to develop your own plugins.
+Icinga Notifications comes packed with channels, but also enables you to develop your own channels.
 
-To make those plugins available to Icinga Notifications, they must be placed in the
+To make those channels available to Icinga Notifications, they must be placed in the
 [channels directory](03-Configuration.md#channels-directory),
 which is being done automatically for package installations.
-Afterwards they can be configured through Icinga Notifications Web.
+At startup, Icinga Notifications scans this directory, starts each channel once to query its configuration options
+and stores these options in the database.
+Using this information, Icinga Notifications Web allows channels to be configured,
+which are then started, configured, and lastly used to send notification events from Icinga Notifications.
 
 ## Technical Channel Description
 
-Channel plugins are processes that run continuously and independently of each other. They receive many requests over
-their lifetime. They receive JSON-formatted requests on stdin and reply with JSON-formatted responses on stdout. The
-request and response structure is inspired by JSON-RPC.
+!!! warning
+
+    As this is still an early preview version, things might change.
+    There may still be severe bugs and incompatible changes may happen without any notice.
+
+Channel plugins are independent processes that run continuously, supervised by Icinga Notifications.
+They receive JSON-formatted requests on `stdin` and reply with JSON-formatted responses on `stdout`.
+The request and response structure is inspired by JSON-RPC.
+
+Note that this documentation uses beautified JSON for ease of reading.
+
+For logging or debugging, channels can write to `stderr`, which will be logged by Icinga Notifications.
 
 ### Request
 
-The request must be in JSON format and should contain following keys:
+A channel receives a request as a JSON object with the following fields:
 
 - `method`: The request method to call.
-- `params`: The params for request method.
+- `params`: Optional params for the request method.
 - `id`: Unsigned int value. Required to assign the response to its request as responses can be sent out of order.
```

Comment:
Hi, 

the field description could be normalized:

`fieldname: description of what it does. optional/madatory. data type`

Regards

Reply:
Yes, maybe as a markdown table, like [Icinga 2](https://icinga.com/docs/icinga-2/latest/doc/12-icinga2-api/#creating-config-objects) does.

Reply:
Thanks., I liked this idea. I tried to unify all fields to:
> name: Short description. Type.

Reply:
I switched to a table, as you suggested @yhabteab. This has the only downside that diffs altering the column width span over the whole table. But, it looks nicer :)

Reply:
@yhabteab: I tried to mimic Icinga 2's layout.